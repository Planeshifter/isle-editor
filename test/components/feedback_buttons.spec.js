// MODULES //

import React from 'react';
import renderer from 'react-test-renderer';
import Provider from '@isle-project/components/internal/provider';
import FeedbackButtons from '@isle-project/components/feedback/main.js';
import Session from '@isle-project/session';


// VARIABLES //

const session = new Session( {}, true );
global.window.URL = {
	createObjectURL() {}
};


// TESTS //

describe( '<FeedbackButtons />', function test() {
	it( 'renders feedback buttons (horizontal format)', () => {
		const buttons = renderer.create(
			<Provider session={session}>
				<FeedbackButtons />
			</Provider>
		).toJSON();
		expect( buttons ).toMatchSnapshot();
	});

	it( 'renders feedback buttons (vertical format)', () => {
		const buttons = renderer.create(
			<Provider session={session}>
				<FeedbackButtons vertical />
			</Provider>
		).toJSON();
		expect( buttons ).toMatchSnapshot();
	});

	it( 'renders feedback buttons without option to supply custom feedback', () => {
		const buttons = renderer.create(
			<Provider session={session}>
				<FeedbackButtons customFeedback={false} />
			</Provider>
		).toJSON();
		expect( buttons ).toMatchSnapshot();
	});

	it( 'renders feedback buttons with custom messages', () => {
		const buttons = renderer.create(
			<Provider session={session}>
				<FeedbackButtons
					confusedMsg="Unclear question"
					feedbackMsg="Critique question wording"
					understoodMsg="Clear question"
				/>
			</Provider>
		).toJSON();
		expect( buttons ).toMatchSnapshot();
	});
});
