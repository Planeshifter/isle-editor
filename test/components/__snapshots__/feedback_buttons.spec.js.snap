// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders feedback buttons (horizontal format) 1`] = `
<span
  className="sk-cube-grid"
>
  <span
    className="sk-cube sk-cube1"
  />
  <span
    className="sk-cube sk-cube2"
  />
  <span
    className="sk-cube sk-cube3"
  />
  <span
    className="sk-cube sk-cube4"
  />
  <span
    className="sk-cube sk-cube5"
  />
  <span
    className="sk-cube sk-cube6"
  />
  <span
    className="sk-cube sk-cube7"
  />
  <span
    className="sk-cube sk-cube8"
  />
  <span
    className="sk-cube sk-cube9"
  />
</span>
`;

exports[`renders feedback buttons (vertical format) 1`] = `
<div
  className="feedback-buttons"
  id="feedback-3"
>
  <div
    className="btn-group-vertical"
    role="group"
    style={
      Object {
        "float": "right",
      }
    }
  >
    <button
      aria-label="Confused"
      className="feedback-button btn btn-light btn-small"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <test-file-stub
        className="feedback-icon"
      />
    </button>
    <button
      aria-label="Understood"
      className="feedback-button btn btn-light btn-small"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <test-file-stub
        className="feedback-icon"
      />
    </button>
    <button
      aria-label="give-custom-feedback"
      className="feedback-button btn btn-light btn-small"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <test-file-stub
        className="feedback-icon"
      />
    </button>
    <div
      id="feedback-3_response_visualizer"
    />
  </div>
  <div
    id="response"
  />
</div>
`;

exports[`renders feedback buttons with custom messages 1`] = `
<div
  className="feedback-buttons"
  id="feedback-5"
>
  <div
    className="btn-group"
    role="group"
    style={
      Object {
        "float": "right",
      }
    }
  >
    <button
      aria-label="Confused"
      className="feedback-button btn btn-light btn-small"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <test-file-stub
        className="feedback-icon"
      />
    </button>
    <button
      aria-label="Understood"
      className="feedback-button btn btn-light btn-small"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <test-file-stub
        className="feedback-icon"
      />
    </button>
    <button
      aria-label="give-custom-feedback"
      className="feedback-button btn btn-light btn-small"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <test-file-stub
        className="feedback-icon"
      />
    </button>
    <div
      id="feedback-5_response_visualizer"
    />
  </div>
  <div
    id="response"
  />
</div>
`;

exports[`renders feedback buttons without option to supply custom feedback 1`] = `
<div
  className="feedback-buttons"
  id="feedback-4"
>
  <div
    className="btn-group"
    role="group"
    style={
      Object {
        "float": "right",
      }
    }
  >
    <button
      aria-label="Confused"
      className="feedback-button btn btn-light btn-small"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <test-file-stub
        className="feedback-icon"
      />
    </button>
    <button
      aria-label="Understood"
      className="feedback-button btn btn-light btn-small"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <test-file-stub
        className="feedback-icon"
      />
    </button>
    <div
      id="feedback-4_response_visualizer"
    />
  </div>
  <div
    id="response"
  />
</div>
`;
